### ISSUE TITLE
Endless Loop

### ISSUE THREAD

**Issue by kpirnie** [sonroyaalmerol/m3u-stream-merger-proxy#84]
## Describe the bug

Getting an endless loop with a few "channels".  Just getting this message over and over
```
2024/08/19 09:15:20 Proxying 192.168.2.20:59276 to http://my_provider/the_channel_THIRD.m3u8
2024/08/19 09:15:20 Current number of connections for M3U_1: 1
2024/08/19 09:15:20 Stream ended (EOF reached): 192.168.2.20:59276
2024/08/19 09:15:20 Exit code 1 received from http://my_provider/the_channel_THIRD.m3u8
2024/08/19 09:15:20 Server connection failed: http://my_provider/the_channel_THIRD.m3u8
2024/08/19 09:15:20 Retrying other servers...
2024/08/19 09:15:20 Current number of connections for M3U_1: 0
2024/08/19 09:15:20 Current number of connections for M3U_1: 0
```
I have 3 separate streams to this specific channel, none of the others get "tried", and this one is actually the last in the list of them.
Example:
```
#EXTINF:-1 tvg-id="0.0-z - Roseanne" tvg-chno="0.0" tvg-name="z - Roseanne", z - Roseanne
http://my_provider/the_channel_FIRST.m3u8
#EXTINF:-1 tvg-id="0.0-z - Roseanne" tvg-chno="0.0" tvg-name="z - Roseanne", z - Roseanne
http://my_second_provider/the_channel_SECOND
#EXTINF:-1 tvg-id="0.0-z - Roseanne" tvg-chno="0.0" tvg-name="z - Roseanne", z - Roseanne
http://my_provider/the_channel_THIRD.m3u8
```

**Comment by aniel300** [comment]
@kpirnie i also got an error like this the other day. hopefully a fix can be implemented 🙏

**Comment by sonroyaalmerol** [comment]
I've added some logic to track the failed M3Us so it doesn't attempt to try them again. Can you try them out on the latest dev build?

**Comment by aniel300** [comment]
is there a way to implement a retry option?

**Comment by sonroyaalmerol** [comment]
Yes, i'm planning to implement retries with customizable max retries after i've confirmed the current solution to be working.

**Comment by aniel300** [comment]
you are a God sent.

**Comment by aniel300** [comment]
@sonroyaalmerol here are my logs in case it is of any help, thanks.
https://github.com/sonroyaalmerol/m3u-stream-merger-proxy/issues/72#issuecomment-2285492825

**Comment by kpirnie** [comment]
> Yes, i'm planning to implement retries with customizable max retries after i've confirmed the current solution to be working.

i am unable to test due to issue #89

**Comment by kpirnie** [comment]
Issue appears to be corrected

**Comment by aniel300** [comment]
was there an update that fixed this?

**Comment by kpirnie** [comment]
> was there an update that fixed this?

Not sure.  It's possible the fix for the WRONGTYPE also had fixes for this tho

**Comment by aniel300** [comment]
ahhh ok

**Comment by kpirnie** [comment]
So... while the streams are back working again, it does not seem that the MAX_RETRIES is.  Hit an endless loop with one of my non-working channels today

```
2024/08/24 06:53:53 Current number of connections for M3U_1: 0
2024/08/24 06:53:53 Current number of connections for M3U_1: 0
2024/08/24 06:53:54 [DEBUG] Successfully fetched stream from http://myprovider/1356094.m3u8
2024/08/24 06:53:54 Proxying 192.168.2.20:56427 to http://myprovider/1356094.m3u8
2024/08/24 06:53:54 Current number of connections for M3U_1: 1
2024/08/24 06:53:54 Stream ended (EOF reached): 192.168.2.20:56427
2024/08/24 06:53:54 Exit code 1 received from http://myprovider/1356094.m3u8
2024/08/24 06:53:54 Retrying other servers...
2024/08/24 06:53:54 [DEBUG] Stream attempt 1 out of 5
2024/08/24 06:53:54 Skipping M3U_1: marked as previous stream
2024/08/24 06:53:54 [DEBUG] All streams skipped in lap 0
2024/08/24 06:53:54 [DEBUG] Stream attempt 2 out of 5
```

Repeats until I change the "channel".

**Comment by sonroyaalmerol** [comment]
Can I confirm if this only happens on `.m3u8` streams?

**Comment by kpirnie** [comment]
> Can I confirm if this only happens on `.m3u8` streams?

I'll see if I have any that are not m3u or m3u8 and get back to ya :)

**Comment by kpirnie** [comment]
It does appear to be only the m3u8 streams that are doing it.

**Comment by sonroyaalmerol** [comment]
I just merged the fix to properly handle m3u files. It should be up in the dev image now

**Comment by aniel300** [comment]
m3u or m3u8?

**Comment by sonroyaalmerol** [comment]
Should include both

**Comment by aniel300** [comment]
ok perfect and thanks

**Comment by kpirnie** [comment]
There we go.  Perfection :)

**Comment by gsanguinetti** [comment]
Hi! I've reached out to this error too. I hope you can include it in a release version soon so I can pick it up in my arm64 docker (dev image is only available for amd64).

**Comment by sonroyaalmerol** [comment]
Should be included in `0.12.1`. I just triggered the build and will be done in a few minutes.
